### ISSUE TITLE
[Bug]: Can't instantiate FabricImage with node-canvas Image since v6

### ISSUE THREAD

**Issue by SvenFackert** [fabricjs/fabric.js#10032]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.2

### In What environments are you experiencing the problem?

Node.js

### Node Version (if applicable)

20.0.0

### Link To Reproduction

Description should be sufficient (references to specific code given)

### Steps To Reproduce

1. Create an Image using `node-canvas`
2. Pass it to the `FabricImage` constructor

```
import { Image } from 'canvas';

const nodeCanvasImage = new Image();
const fabricImage = new FabricImage(nodeCanvasImage);
```

### Expected Behavior

Similar to [this bug report](https://github.com/fabricjs/fabric.js/issues/9586) I would expect, that passing a node-canvas Image object still works. We were using fabric.js together with `node-canvas` for our node backends as previously described in the docs [here](http://fabricjs.com/fabric-intro-part-4#node). It would be nice to be able to use the node-canvas Image without stubbing/monkey patching the `classList` property as before. And also without adding `js-dom` just for this use case. What do you think?

### Actual Behavior

The constructor of FabricImage will call `Image.setElement()`, which tries to call the `classList.add()` method on the passed image. The classList is only supported by `HTMLImageElement` and not node-canvas Image. This will result in the following runtime error: `Cannot read properties of undefined (reading 'add')`.

**Comment by asturur** [comment]
We keep JSDOM for this purpose.
are you importing fabricImage from 'fabric/node' ? or from 'fabric' ?

**Comment by zhangweiwei0326** [comment]
+1 node_modules\fabric\dist\index.node.cjs:24117
    element.classList.add(FabricImage.CSS_CANVAS);
                      ^
TypeError: Cannot read properties of undefined (reading 'add')

**Comment by bkeepers** [comment]
Workaround for now is to add a `classList` property with an `add` merthod:

```diff
  import { Image } from 'canvas';

  const img = new Image();
+ img.classList = new Set();
  const fabricImage = new FabricImage(img);
```

**Comment by asturur** [comment]
I don't even know why we add that  class. Like if you want to target some css on that element, but why?
Seems dangerous especially if you are reusing an element that is also on the dom and you are referencing it adding a class with attached style ( not provided by default ) is going to mess up with how that element look like.

I ll flag that for removal, i really can't understand the usefulness of it

**Comment by asturur** [comment]
I deprecated it.

**Comment by SvenFackert** [comment]
Hi everyone, thanks for tackling this issue! I appreciate it 🙏.  Please let me know, when this change is published, so we can update fabric.js in our projects 👍

**Comment by SvenFackert** [comment]
> 'fabric/node'

From 'fabric/node'

**Comment by SvenFackert** [comment]
> We keep JSDOM for this purpose. are you importing fabricImage from 'fabric/node' ? or from 'fabric' ?

From 'fabric/node'

**Comment by asturur** [comment]
It will take time before is published because in theory is a breaking change.
What we can do meantime is add an optional chainer to make it work

**Comment by asturur** [comment]
Do you feel like you can do it and open a PR? is just a `?` between the call and the parenthesis

**Comment by SvenFackert** [comment]
@asturur yes, sure. Here it is: https://github.com/fabricjs/fabric.js/pull/10412 👍 It's not urgent because I already monkey patched that in our project. But it might help others till the classList access gets removed in the future 👍

**Comment by asturur** [comment]
Yes but monkeypatching for those small things is evil.
If the fix makes sense better to have it.
Most of my effort are toward monkepatching be avoidable with apis and subclassing
